test/CMakeLists.txt

```diff
@@ -54,22 +54,19 @@ if(JSON_Coverage)
 endif()
 
 #############################################################################
-# Catch library with the main function to speed up build
+# doctest library with the main function to speed up build
 #############################################################################
 
-add_library(catch_main OBJECT
+add_library(doctest_main OBJECT
     "src/unit.cpp"
 )
-set_target_properties(catch_main PROPERTIES
+set_target_properties(doctest_main PROPERTIES
     COMPILE_DEFINITIONS "$<$<CXX_COMPILER_ID:MSVC>:_SCL_SECURE_NO_WARNINGS>"
     COMPILE_OPTIONS "$<$<CXX_COMPILER_ID:MSVC>:/EHsc;$<$<CONFIG:Release>:/Od>>"
 )
-if (${CMAKE_VERSION} VERSION_LESS "3.8.0") 
-    target_compile_features(catch_main PUBLIC cxx_range_for)
-else()
-    target_compile_features(catch_main PUBLIC cxx_std_11)
-endif()
-target_include_directories(catch_main PRIVATE "thirdparty/catch")
+
+target_compile_features(doctest_main PUBLIC cxx_std_11)
```

Comment:
Are you sure you do not need the `if` here? This may be an issue for CMake versions prior 3.8.0.

Reply:
You are correct - I assumed initially that the ```3.8``` check was for ```cxx_range_for``` but it was indeed if ```cxx_std_11``` is supported. Adding it back again.

Reply:
Actually it was without that check up until commit 4fd9b52fc2ff796e2dcf207387f7a3b217d5e314 and it used to work - guess people didn't try the tests with a CMake version older than 3.8.

For doctest 2.x C++11 is actually mandatory. For new compilers such as gcc 7/8 C++11 is on by default, but for older ones like 4.x/5 (and maybe 6) C++11 has to be explicitly stated. So I'll add ```-std=c++0x``` to the ```CXX_FLAGS``` property for the target if cmake < 3.8

Reply:
I get it - ```cxx_range_for``` is not for that particular feature but for "c++0x" support in general - and was one of the first C++11 features to be added to compilers so ```cxx_range_for``` was added as a compile feature long before ```cxx_std_11```...... So I'll just revert the code to have that same exact if/else so I don't have to deal with ```CXX_STANDARD```